File: uva online judge/00352 - The Seasonal War.py
import sys

def read_test_case():
    line = sys.stdin.readline().strip()
    if line == '':
        return None
    n = int(line)
    grid = []
    for i in range(n):
        grid.append([int(i) for i in sys.stdin.readline().strip()])
    return n, grid

def fill(grid, n, x, y):
    if x < 0 or x >= n or y < 0 or y >= n or grid[x][y] != 1:
        return
    grid[x][y] = 2
    for dx, dy in ((-1,0),(-1,1),(0,1),(1,1),(1,0),(1,-1),(0,-1),(-1,-1)):
        fill(grid, n, x + dx, y + dy)

if __name__ == '__main__':
    i = 1
    for n, grid in iter(read_test_case, None):
        eagles_cnt = 0
        for x in range(n):
            for y in range(n):
                if grid[x][y] == 1:                    
                    fill(grid, n, x, y)
                    eagles_cnt += 1
        print('Image number {} contains {} war eagles.'.format(i, eagles_cnt))
        i += 1

File: uva online judge/00352 - The Seasonal War.c
#include <stdio.h>
#include <string.h>
#define MAX_LENGTH 25

int dx[] = {1,1,0,-1,-1,-1, 0, 1};
int dy[] = {0,1,1, 1, 0,-1,-1,-1};

void fill(char grid[MAX_LENGTH][MAX_LENGTH], int n, int x, int y){
    int i;
    if(x < 0 || x >= n || y < 0 || y >= n || grid[x][y] != '1')
        return;
    grid[x][y] = '0';
    for(i = 0; i < 8; i++){
        fill(grid, n, x + dx[i], y + dy[i]);
    }
}

int main(){
    int x, y, n, cnt, case_i = 1;
    char grid[MAX_LENGTH][MAX_LENGTH], buffer[MAX_LENGTH + 2];
    while(fgets(buffer, sizeof(buffer), stdin)){
        sscanf(buffer, "%d", &n);
        for(x=0; x<n; x++){
            fgets(buffer, sizeof(buffer), stdin);
            memcpy(grid[x], buffer, sizeof(char) * n);
        }
        cnt = 0;
        for(x = 0; x < n; x++){
            for(y = 0; y < n; y++){
                if(grid[x][y] == '1'){
                    fill(grid, n, x, y);
                    cnt++;
                }
            }
        }
        printf("Image number %d contains %d war eagles.\n", case_i, cnt);
        case_i++;
    }
    return 0;
}